docs/pages/enroll-resources/database-access/enroll-self-hosted-databases/oracle-exadata.mdx

```diff
@@ -0,0 +1,411 @@
+
+```
+SSL_CLIENT_AUTHENTICATION = TRUE
+SQLNET.AUTHENTICATION_SERVICES = (ALL)
+```
+
+For example, if there is a database named `spark`, the `oracle` user should have access to the automatically generated `/home/oracle/spark.env` file, which contains database-specific environment variables. After sourcing this file, the path `$ORACLE_HOME/network/admin/sqlnet.ora` will point to the `sqlnet.ora` file for the `spark` database and its associated Oracle home.
+
+```code
+# Load environment variables for the 'spark' database
+$ . spark.env 
+
+# Edit the configuration file at $ORACLE_HOME/network/admin/sqlnet.ora
+...
+```
+
+Repeat these steps for each additional database or database home as required.
+
+### Configure Oracle user account
+
+Your Oracle user accounts must be configured to require a valid client certificate. 
+
+```sql
+ALTER USER alice IDENTIFIED EXTERNALLY AS 'CN=alice';
+```
+
+If you're creating a new user:
+
+```sql
+CREATE USER alice IDENTIFIED EXTERNALLY AS 'CN=alice';
+GRANT CREATE SESSION TO alice;
+```
+
+### Trust Teleport Database Client CA
+
+Teleport uses mutual TLS authentication with Oracle Exadata. It must be configured with Teleport's certificate authority to be able to verify client certificates.
+
+Export the Teleport Database Client CA on your local machine and copy it to target Oracle Exadata VM.
+
+```code
+# Export Teleport's database client certificate authority
+$ tctl auth export --type=db-client > teleport-db-client-ca.crt
+# Copy the CA to the Oracle Exadata VM
+$ scp teleport-db-client-ca.crt opc@<Var name="demodb-vm" />:/tmp/teleport-db-client-ca.crt
+```
+
+As the `grid` user, update the grid TCPS wallet to trust the Teleport User Database CA.
+
+```code
+# Read the wallet password
+$ mkstore -wrl /u01/app/oracle/admin/cprops/cprops_wallet -nologo -viewEntry grid_tcps_wallet_passwd
+grid_tcps_wallet_passwd = <wallet password>
+# Update the grid TCPS wallet; provide the password when prompted
+$ orapki wallet add -wallet "/var/opt/oracle/dbaas_acfs/grid/tcps_wallets" -trusted_cert -cert /tmp/teleport-db-client-ca.crt
+```
+
+### Enable TLS auth for grid listener
+
+Adjust your grid listener configuration at `$ORACLE_HOME/network/admin/listener.ora` to enable TLS auth with the following entry:
+
+```
+SSL_CLIENT_AUTHENTICATION = TRUE
+```
+
+### Restart the listener
+
+Once finished, restart the listener.
+
+```code
+$ lsnrctl stop
+$ lsnrctl start
+```
+
+## Step 2/6. Collect database configuration data.
+
+Export the built-in database certificate from Oracle. Teleport will use this certificate to verify the database connection.
+
+```code
+# Export the database certificate. Update the "-dn" parameter to match your actual setup.
+$ orapki wallet export -wallet "/var/opt/oracle/dbaas_acfs/grid/tcps_wallets" -dn "CN=<Var name="demodb-vm-scan.exadatadomain.oke.oraclevcn.com" />" -cert /tmp/oracle-server-certificate.crt
```

Comment:
You can probably just peek at the wallet with:

```
$ orapki wallet display -wallet "/var/opt/oracle/dbaas_acfs/grid/tcps_wallets"
```

But it would be good to test on a live server first, which we don't have right now.